packages/grid/x-data-grid/src/components/GridRow.tsx

```diff
@@ -248,14 +250,28 @@ function GridRow(props: React.HTMLAttributes<HTMLDivElement> & GridRowProps) {
     [apiRef, onClick, publish, rowId],
   );
 
+  const sizes = apiRef.current.unstable_getRowInternalSizes(rowId);
+
+  let minHeight = rowHeight;
+  if (minHeight === 'auto' && sizes && Object.keys(sizes).length > 1) {
+    const maximumSize = Object.entries(sizes).reduce((acc, [key, size]) => {
+      if (!/^base[A-Z]/.test(key) || size <= acc) {
+        return acc;
+      }
+      return size;
```

Comment:
```suggestion
      const isBaseHeight = /^base[A-Z]/.test(key);
      if (isBaseHeight && size > acc) {
        return size;
      }
      return acc;
```

To make it more readable